ciphey/basemods/Crackers/XandY.py

```diff
@@ -36,37 +39,63 @@ def getTarget() -> str:
 
     def attemptCrack(self, ctext: str) -> List[CrackResult]:
         """
-        Checks an input if it only consists of two different letters.
-        If this is the case, it attempts to regard those two letters as
-        0 and 1 and then converts it to ascii text.
+        Checks an input if it only consists of two or three different letters.
+        If this is the case, it attempts to regard those letters as
+        0 and 1 (with the third characters as an optional delimiter) and then 
+        converts it to ascii text.
         """
         logger.trace("Attempting X-Y replacement.")
         variants = []
         candidates = []
         result = []
-        ctext = ctext.lower().replace(" ", "").strip()
 
-        # cset contains every unique value in the cstring
+        # Convert the ctext to all-lowercase and regex-match & replace all whitespace
+        ctext = ctext.lower()
+        ctext = re.sub(r"\s+", "", ctext, flags=re.UNICODE)
+
+        # cset contains every unique value in the ctext
         cset = list(set(list(ctext)))
-        if len(cset) != 2:
```

Comment:
We should store the length by itself in a var, `cset_len` since we repeatedly call the function `len()`, saving us nanoseconds of time :)